packages/filesystem/src/common/files.ts

```diff
@@ -558,85 +558,291 @@ export function ensureFileSystemProviderError(error?: Error): Error {
+ * local disk as well as remote locations like ftp-servers, REST-services etc. A {@link FileSystemProvider} is registered for a certain
+ * scheme and can handle all resources whose uri does conform to that scheme.
+ */
 export interface FileSystemProvider {
 
+    /** The {@link FileSystemProviderCapabilities} for this provider. */
     readonly capabilities: FileSystemProviderCapabilities;
+
+    /** * Event that is fired if the capabilities of this provider have changed. */
     readonly onDidChangeCapabilities: Event<void>;
 
+    /** Event that is fired if a (watched) file in the filesystem of this provider has changed. */
     readonly onDidChangeFile: Event<readonly FileChange[]>;
+
+    /** Event that is fired if an error occurred when watching files in the filesystem of this provider. */
     readonly onFileWatchError: Event<void>;
+
+    /**
+     * Watch the given resource and react to changes by firing the {@link FileSystemProvider#onDidChangeFile} event.
+     * @param resource `URI` of the resource to be watched.
+     * @param opts Options to define if the resource should be watched recursively and to
+     *  provide a set of resources that should be excluded from watching.
+     *
+     * @returns A `Disposable` that can be invoked to stop watching the resource.
+     */
     watch(resource: URI, opts: WatchOptions): IDisposable;
 
+    /**
+     * Retrieve metadata about a given file.
+     *
+     * @param uri The `URI` of the file to retrieve meta data about.
+     * @returns A promise of the metadata about the resource.
+     */
     stat(resource: URI): Promise<Stat>;
+
+    /**
+     * Create a new directory using the given resource uri.
+     * @param resource The `URI` of the new folder.
+     */
     mkdir(resource: URI): Promise<void>;
+
+    /**
+     * Retrieve the content of a given directory.
+     * @param resource The `URI` of the directory.
+     *
+     * @returns A map containing the {@link FileType} for each child resource (uri).
+     */
     readdir(resource: URI): Promise<[string, FileType][]>;
+
+    /**
+     * Delete the given resource.
+     * @param resource The `URI` of the resource to delete.
+     * @param opts Options to define if files should be deleted recursively and if the trash should be used.
+     */
     delete(resource: URI, opts: FileDeleteOptions): Promise<void>;
 
+    /**
+     * Rename a file or folder.
+     * @param from `URI` of the existing file or folder.
+     * @param to `URI` of the target location.
+     * @param opts Options to define if existing files should be overwritten.
+     */
+
     rename(from: URI, to: URI, opts: FileOverwriteOptions): Promise<void>;
+
+    /**
+     * Optional function that has to be implemented by {@link FileSystemProviderWithFileFolderCopyCapability}.
+     * see {@link  FileSystemProviderWithFileFolderCopyCapability#copy}} for additional documentation.
```

Comment:
There are extra _whitespaces_ after `@link` for the following methods.
Please ignore if these are done on purpose.

Reply:
I'm pretty sure that's a mistake ;)